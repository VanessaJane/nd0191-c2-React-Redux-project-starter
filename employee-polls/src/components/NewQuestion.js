import { useState } from "react";
import { handleAddQuestion } from "../actions/questions";
import { connect } from "react-redux";

const NewQuestion = ({ dispatch }) => {
  const [firstOption, setFirstOption] = useState("");
  const [secondOption, setSecondOption] = useState("");

  const handleInputChange = (event) => {
    event.preventDefault();
    const target = event.target;
    if (target.name === "firstOption") {
      setFirstOption(target.value);
    } else {
      setSecondOption(target.value);
    }
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    dispatch(handleAddQuestion(firstOption, secondOption));
    setFirstOption("");
    setSecondOption("");
  };

  return (
    <div className="center_container">
      <h2 className="center">Would You Rather</h2>
      <h4 className="center">Create Your Own Poll</h4>
      <form onSubmit={handleSubmit} className="question_form">
        <label>
          First Option
          <input
            name="firstOption"
            type="text"
            value={firstOption}
            onChange={handleInputChange}
          />
        </label>
        <label>
          Second Option
          <input
            name="secondOption"
            type="text"
            value={secondOption}
            onChange={handleInputChange}
          />
        </label>
        <button
          type="submit"
          disabled={firstOption === "" || secondOption === ""}
          className="submit_btn"
        >
          Submit
        </button>
      </form>
    </div>
  );
};

export default connect()(NewQuestion);
