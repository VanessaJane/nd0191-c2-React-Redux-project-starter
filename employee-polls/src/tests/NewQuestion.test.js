import { render, screen } from "@testing-library/react";
import store from "../app/store";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import NewQuestion from "../components/NewQuestion";

describe("NewQuestion", () => {
  it("should render the new question page onload", () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <NewQuestion />
        </Provider>
      </MemoryRouter>
    );

    expect(screen.getByText(/Would You Rather/i)).toBeInTheDocument();
    expect(screen.getByText(/Create Your Own Poll/i)).toBeInTheDocument();
    expect(screen.getByText(/First Option/i)).toBeInTheDocument();
    expect(screen.getByText(/Second Option/i)).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    var component = render(
      <MemoryRouter>
        <Provider store={store}>
          <NewQuestion />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
