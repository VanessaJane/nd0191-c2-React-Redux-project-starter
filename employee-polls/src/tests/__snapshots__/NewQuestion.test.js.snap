// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="center_container"
      >
        <h2
          class="center"
        >
          Would You Rather
        </h2>
        <h4
          class="center"
        >
          Create Your Own Poll
        </h4>
        <form
          class="question_form"
        >
          <label>
            First Option
            <input
              name="firstOption"
              type="text"
              value=""
            />
          </label>
          <label>
            Second Option
            <input
              name="secondOption"
              type="text"
              value=""
            />
          </label>
          <button
            class="submit_btn"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="center_container"
    >
      <h2
        class="center"
      >
        Would You Rather
      </h2>
      <h4
        class="center"
      >
        Create Your Own Poll
      </h4>
      <form
        class="question_form"
      >
        <label>
          First Option
          <input
            name="firstOption"
            type="text"
            value=""
          />
        </label>
        <label>
          Second Option
          <input
            name="secondOption"
            type="text"
            value=""
          />
        </label>
        <button
          class="submit_btn"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
